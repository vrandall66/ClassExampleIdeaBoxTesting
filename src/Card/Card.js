import React from "react";
import "./Card.css";

const Card = ({ id, title, description, removeIdea, isFavorite }) => {
  const favoriteClass = isFavorite ? "favorite" : "card";

  return (
    <section className={favoriteClass}>
      <h3>{title}</h3>
      <p>{description}</p>
      <button onClick={() => removeIdea(id)}>ðŸ—‘</button>
    </section>
  );
};

export default Card;
