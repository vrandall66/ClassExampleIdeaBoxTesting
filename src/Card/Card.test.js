import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("Card", () => {
  it("should match the snapshot with all data passed in correctly", () => {
    const wrapper = shallow(
      <Card
        title="Bananas"
        description="B A N A N A S"
        id={3}
        removeIdea={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the favorited snapshot", () => {
    const wrapper = shallow(
      <Card
        title="Bananas"
        description="B A N A N A S"
        id={3}
        removeIdea={jest.fn()}
        isFavorite={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the removeIdea prop with the Card id when clicked', () => {
    const removeIdeaMock = jest.fn();
    const wrapper = shallow(
      <Card
        title="Bananas"
        description="B A N A N A S"
        id={3}
        removeIdea={removeIdeaMock}
        isFavorite={true}
      />
    );
    // Execution
    wrapper.find('button').simulate('click');

    // Expectation
    expect(removeIdeaMock).toHaveBeenCalledWith(3);
  })
});
