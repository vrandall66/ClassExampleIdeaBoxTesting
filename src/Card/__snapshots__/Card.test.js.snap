// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<section
  className="card"
>
  <h3>
    Bananas
  </h3>
  <p>
    B A N A N A S
  </p>
  <button
    onClick={[Function]}
  >
    ðŸ—‘
  </button>
</section>
`;
